test/parallel/market.rpc.check.test.ts

```diff
@@ -0,0 +1,473 @@
+      ).then((result) => {
+        const eventResponse = getEventResultFromMangataTx(result);
+        expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+      });
+
+      const buyPriceAfter = await calculateBuyPriceByMarket(
+        liqId,
+        firstCurrency,
+        threshold.muln(2),
+      );
+
+      const diff = buyPriceAfter.sub(buyPriceWithImpact[1]);
+
+      expect(buyPriceBefore).bnEqual(buyPriceWithImpact[0]);
+      expect(diff).bnLt(buyPriceAfter.muln(3).divn(1000));
+    });
+  },
+);
+
+test("check getBurnAmount", async () => {
+  await Sudo.batchAsSudoFinalized(
+    Assets.mintNative(testUser),
+    Assets.mintToken(firstCurrency, testUser, threshold.muln(20)),
+    Assets.mintToken(secondCurrency, testUser, threshold.muln(20)),
+  );
+
+  await signTx(
+    api,
+    Market.createPool(
+      firstCurrency,
+      threshold.muln(10),
+      secondCurrency,
+      threshold.muln(5),
+    ),
+    testUser.keyRingPair,
+  ).then((result) => {
+    const eventResponse = getEventResultFromMangataTx(result);
+    expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+  });
+  testUser.addAsset(firstCurrency);
+  testUser.addAsset(secondCurrency);
+
+  liqId = await getPoolId(firstCurrency, secondCurrency);
+  const burnAmount = await getBurnAmount(liqId, threshold);
+
+  await testUser.refreshAmounts(AssetWallet.BEFORE);
+
+  await signTx(
+    api,
+    Market.burnLiquidity(liqId, threshold),
+    testUser.keyRingPair,
+  ).then((result) => {
+    const eventResponse = getEventResultFromMangataTx(result);
+    expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+  });
+
+  await testUser.refreshAmounts(AssetWallet.AFTER);
+
+  const firstCurrencyDiff: any = testUser
+    .getAsset(firstCurrency)
+    ?.amountAfter.free.sub(
+      testUser.getAsset(firstCurrency)?.amountBefore.free!,
+    );
+
+  const secondCurrencyDiff: any = testUser
+    .getAsset(secondCurrency)
+    ?.amountAfter.free.sub(
+      testUser.getAsset(secondCurrency)?.amountBefore.free!,
+    );
+
+  expect(burnAmount.firstTokenAmount).bnEqual(firstCurrencyDiff);
+  expect(burnAmount.secondTokenAmount).bnEqual(secondCurrencyDiff);
+});
+
+test("check calculation for minting", async () => {
+  await Sudo.batchAsSudoFinalized(
+    Assets.mintNative(testUser),
+    Assets.mintToken(firstCurrency, testUser, threshold.muln(20)),
+    Assets.mintToken(secondCurrency, testUser, threshold.muln(20)),
+  );
+
+  await signTx(
+    api,
+    Market.createPool(
+      firstCurrency,
+      threshold.muln(10),
+      secondCurrency,
+      threshold.muln(5),
+    ),
+    testUser.keyRingPair,
+  ).then((result) => {
+    const eventResponse = getEventResultFromMangataTx(result);
+    expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+  });
+
+  liqId = await getPoolId(firstCurrency, secondCurrency);
+
+  testUser.addAsset(liqId);
+  await testUser.refreshAmounts(AssetWallet.BEFORE);
```

Comment:
Ive seen this more than 3 times already :)  plz refactor

Reply:
There was difference between the code in the different tests, but I refactored it and created one function for all type